kotlinx-coroutines-core/common/src/EventLoop.common.kt

```diff
@@ -428,53 +430,53 @@ internal abstract class EventLoopImplBase: EventLoopImplPlatform(), Delay {
             return when {
-                dTime > 0 -> 1
-                dTime < 0 -> -1
+                dTime > Duration.ZERO -> 1
+                dTime < Duration.ZERO -> -1
                 else -> 0
             }
         }
 
-        fun timeToExecute(now: Long): Boolean = now - nanoTime >= 0L
+        fun timeToExecute(now: ComparableTimeMark): Boolean = now - timeMark >= Duration.ZERO
 
-        fun scheduleTask(now: Long, delayed: DelayedTaskQueue, eventLoop: EventLoopImplBase): Int = synchronized<Int>(this) {
+        fun scheduleTask(now: ComparableTimeMark, delayed: DelayedTaskQueue, eventLoop: EventLoopImplBase): Int = synchronized<Int>(this) {
             if (_heap === DISPOSED_TASK) return SCHEDULE_DISPOSED // don't add -- was already disposed
             delayed.addLastIf(this) { firstTask ->
                 if (eventLoop.isCompleted) return SCHEDULE_COMPLETED // non-local return from scheduleTask
                 /**
                  * We are about to add new task and we have to make sure that [DelayedTaskQueue]
                  * invariant is maintained. The code in this lambda is additionally executed under
-                 * the lock of [DelayedTaskQueue] and working with [DelayedTaskQueue.timeNow] here is thread-safe.
+                 * the lock of [DelayedTaskQueue] and working with [DelayedTaskQueue.timeMark] here is thread-safe.
                  */
                 if (firstTask == null) {
                     /**
-                     * When adding the first delayed task we simply update queue's [DelayedTaskQueue.timeNow] to
+                     * When adding the first delayed task we simply update queue's [DelayedTaskQueue.timeMark] to
                      * the current now time even if that means "going backwards in time". This makes the structure
                      * self-correcting in spite of wild jumps in `nanoTime()` measurements once all delayed tasks
                      * are removed from the delayed queue for execution.
                      */
-                    delayed.timeNow = now
+                    delayed.timeMark = now
                 } else {
                     /**
-                     * Carefully update [DelayedTaskQueue.timeNow] so that it does not sweep past first's tasks time
+                     * Carefully update [DelayedTaskQueue.timeMark] so that it does not sweep past first's tasks time
                      * and only goes forward in time. We cannot let it go backwards in time or invariant can be
                      * violated for tasks that were already scheduled.
                      */
-                    val firstTime = firstTask.nanoTime
+                    val firstTime = firstTask.timeMark
                     // compute min(now, firstTime) using a wrap-safe check
-                    val minTime = if (firstTime - now >= 0) now else firstTime
+                    val minTime = if (firstTime - now >= Duration.ZERO) now else firstTime
```

Comment:
I am not sure is such wrap-safe checks are still relevant.
Is it safe to transform such expressions to `firstTime >= now` or `minOf(firstTime, now)`?